AudioStream.js

```diff
@@ -97,29 +98,44 @@ module.exports = class AudioStream {
         }
     }
 
-
-    async extractVideoIds(url) {
+    async extractVideoIdsFromPlaylist(playlistUrl){
+        
         const s = new Set();
-        try {
-            let temp = await fetch(url).then(res => res.text());
-            while (temp !== "") {
-                const ind = temp.indexOf('videoId');
-                if (temp.substring(ind, ind + 40).includes("videoId")) {
-                    const videoIdStart = ind + 10;
-                    const videoIdEnd = ind + 23;
-                    const videoId = temp.substring(videoIdStart, videoIdEnd).replace(/["\[\],]/g, "");
-                    console.log(temp.substring(ind, ind + 40))
-                    s.add("https://www.youtube.com/watch?v=" + videoId);
-                }
-                temp = temp.substring(ind + 21);
-            }
-        } catch (error) {
-            console.error('Error:', error);
-        }
-        for (const video of Array.from(s)) {
-            this.queueUp(video)
-        };
-    }
+        const urlParams = new URLSearchParams(new URL(playlistUrl).search);
+        const playlistId = urlParams.get('list');
+      
+        let nextPageToken = '';
```

Comment:
add a comment here explaining what this is doing for example

```js
// the below extracts the `list` param from the url of a playlist url,
// for example the value extracted from below would be PLxLkp2u5doGjtWoc577n_1PHK8Iiu7oHO
// https://www.youtube.com/watch?v=XjhzXzaikpU&list=PLxLkp2u5doGjtWoc577n_1PHK8Iiu7oHO
```

Reply:
Done